crates/uv-client/src/base_client.rs

```diff
@@ -439,9 +452,18 @@ impl RetryableStrategy for UvRetryableStrategy {
 /// Check for additional transient error kinds not supported by the default retry strategy in `reqwest_retry`.
 ///
 /// These cases should be safe to retry with [`Retryable::Transient`].
-fn is_extended_transient_error(res: &Result<Response, reqwest_middleware::Error>) -> bool {
-    // Check for connection reset errors, these are usually `Body` errors which are not retried by default.
-    if let Err(reqwest_middleware::Error::Reqwest(err)) = res {
+pub(crate) fn is_extended_transient_error(err: &dyn Error) -> bool {
+    if let Some(err) = find_source::<reqwest_middleware::Error>(&err) {
+        if let Some(io) = find_source::<std::io::Error>(&err) {
+            if io.kind() == std::io::ErrorKind::ConnectionReset
+                || io.kind() == std::io::ErrorKind::UnexpectedEof
+            {
+                return true;
+            }
+        }
+    }
+
+    if let Some(err) = find_source::<reqwest::Error>(&err) {
```

Comment:
Should this just be an `or_else` on `if let Some(err) = find_source::<reqwest_middleware::Error>(&err) `?

Reply:
Conceptually yes but the compiler doesn't allow it, yeah.